octavia_f5/controller/worker/set_queue.py

```diff
@@ -31,6 +31,9 @@ def put_priority(self, item):
         """Add an item to the priority queue."""
         self.priority_queue.add(item)
         self.queue.discard(item)
+        # notify polling threads
+        with self.not_empty: # acquire self.mutex for self.not_empty
+            self.not_empty.notify()
```

Comment:
I'm not sure that this is gonna work because "original" method `_put` was working without this notification. How did you test that?

Reply:
Ok, read the source of the module, and I got the point.

Reply:
Yeah, sorry, I should have linked to it.
For the record and anyone reading in the future: The inner `_put` method is called by the upstream (non-overriden) `put` method [here](https://github.com/python/cpython/blob/3.13/Lib/queue.py#L176) before that upstream `put` method calls `self.not_empty.notify()` (with the mutex locked by the `with` scope), which is what was missing in `put_priority`.